store/sqlstore/supplier_test.go

```diff
@@ -19,6 +19,33 @@ import (
 	"github.com/mattermost/mattermost-server/v5/store/storetest"
 )
 
+func TestSupplierLicenseRace(t *testing.T) {
```

Comment:
How does this test fail?

Reply:
Now it won't, hopefully. 
I've used it to consistently reproduce the race. Before the changes it would fail when using the `-race` flag.

Reply:
Oh I see. Makes sense. Would you add a comment for that?

Reply:
done :+1: